lib/open_api_2_json_schema.rb

```diff
@@ -35,43 +37,51 @@ def self.build(type)
 
   def convert_from_file(path)
     data = File.read(path)
-    convert(YAML.load(data))
+    convert(YAML.safe_load(data))
   end
 
   def convert(schema)
     convert_schema(schema).tap do |json_schema|
-      json_schema['$schema'] = 'http://json-schema.org/draft-04/schema#'
+      json_schema["$schema"] = "http://json-schema.org/draft-04/schema#"
     end.to_json
   end
 
   def convert_schema(schema)
     json_schema = schema.except(*NOT_SUPPORTED)
 
-    schema.slice(STRUCTS).each do |struct, data|
+    schema.slice(*STRUCTS).each do |struct, data|
       case data
       when Array
         json_schema[struct] = data.map do |item|
           convert_schema(item)
         end
       when Hash
-        json_schema[struct] = convert_schema(data)
+        if struct == 'allOf' && data['$ref']
+          base_schema = JSON.parse(convert_from_file(data['$ref']))
+          base_schema = base_schema['schema'] if base_schema['schema']
```

Comment:
In which case `schema` field does not exist in `base_schema`?

Reply:
Hmmmmm actually i'm not sure, the sample code from swagger does not have `schema` in it, but I guess this gem is only for dealing with our specific case, so we can assume that `schema` presents all the time 👍

Reply:
Let's not assume that the gem only deals with one specific case. 

So if `schema` field does not exists, we'll merge the whole hash, otherwise, we only merge the hash inside `schema`?

Reply:
That's what the current logic is doing, right? Oh you meant a different thing? 🤔

Reply:
No, I just want to confirm that is what we currently have.

Could you check what is the output of https://github.com/openapi-contrib/openapi-schema-to-json-schema and try to match it?

Reply:
It seems like they doesn't support schema inheritance syntax either

<img width="1342" alt="image" src="https://user-images.githubusercontent.com/83625949/191474763-36bd25ca-25f1-4456-a25d-bf8e1f45892f.png">

<img width="446" alt="image" src="https://user-images.githubusercontent.com/83625949/191474940-89886676-cabe-4fbd-9d7f-c5b51ea2a738.png">

Reply:
I think there is a note here: 

> NOTE: $refs are not handled in any way, so please use a resolver such as [json-schema-ref-parser](https://github.com/APIDevTools/json-schema-ref-parser) or [swagger-cli bundle](https://www.npmjs.com/package/swagger-cli) prior to using this package.

I guess to comply with the standard, we need to implement a ref parser similar to https://github.com/APIDevTools/json-schema-ref-parser, then in FE, we need to handle `allOf` logic instead. Could we check with Tim on whether handling `allOf` is possible on his side?

Reply:
No, an example of the expected result is in https://github.com/hieuk09/open_api_2_json_schema/pull/1#discussion_r976462712. 

We should have this format:
```json
{
  "allOf": [
    {
      # schema 1 
    },
    {
      # schema 2
    }
  ]
}
```

For this gem, I think implementing the ref parser is enough. As I check the FE code, it's not easy to change to support `allOf`, so we can deep merge all schemas in `allOf` within the application instead.